backend/lucene/src/main/java/org/hibernate/search/backend/lucene/search/query/impl/LuceneSearchQueryImpl.java

```diff
@@ -148,6 +153,26 @@ public Sort luceneSort() {
 		return luceneSort;
 	}
 
+	SearchScrollResult<H> next(HibernateSearchMultiReader indexReader, Integer offset, Integer limit) {
```

Comment:
I'd move this method to scroll. If feels weird to have this back-reference to the query.